File: 백준/Silver/1920. 수 찾기/수 찾기.py
def binary_sort(N):
    left_idx = 0
    right_idx = num - 1
    while left_idx <= right_idx:
        mid = (left_idx + right_idx) // 2
        if N > res[mid]:
            left_idx = mid + 1
        elif N < res[mid]:
            right_idx = mid - 1
        else: # 값을 찾은 경우
            return 1
    return 0 # 값을 찾지 못한 경우
        

num = int(input())
res = list(map(int, input().split()))
res.sort()

M = int(input())
cmp = list(map(int, input().split()))

for i in cmp:
    print(binary_sort(i))

File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int binary_sort(int N, const vector<int>& res) {
    int left_idx = 0;
    int right_idx = res.size() - 1;
    while (left_idx <= right_idx) {
        int mid = (left_idx + right_idx) / 2;
        if (N > res[mid])
            left_idx = mid + 1;
        else if (N < res[mid])
            right_idx = mid - 1;
        else // 값을 찾은 경우
            return 1;
    }
    return 0; // 값을 찾지 못한 경우
}

int main() {
    int num;
    cin >> num;

    vector<int> res(num);
    for (int i = 0; i < num; ++i) {
        scanf("%d", &res[i]);
    }
    sort(res.begin(), res.end());

    int M;
    cin >> M;

    vector<int> cmp(M);
    for (int i = 0; i < M; ++i) {
        scanf("%d", &cmp[i]);
    }

    for (int i = 0; i < M; ++i) {
        printf("%d\n", binary_sort(cmp[i], res));
    }

    return 0;
}